wdl-lint/src/util.rs

```diff
@@ -100,6 +104,40 @@ pub fn strip_newline(s: &str) -> Option<&str> {
     s.strip_suffix("\r\n").or_else(|| s.strip_suffix('\n'))
 }
 
+/// Finds the nearest rule ID to the given unknown rule ID,
+/// or `None` if no rule ID is close enough.
+pub fn find_nearest_rule(unknown_rule_id: &str) -> Option<&'static str> {
+    // Collect all known rule IDs
+    let all_rule_ids: Vec<&'static str> = RULE_MAP
+        .keys()
+        .copied()
+        .chain(RESERVED_RULE_IDS.iter().copied())
+        .collect();
```

Comment:
Does this need to be collected?

At first glance, it seems like we could simply iterate over `RULE_MAP.keys().chain(RESERVED_RULE_IDS.iter())` and the length for the threshold would be `RULE_MAP.len() + RESERVED_RULE_IDS.len()`.

Reply:
Shouldn't the threshold be calculated based on the length of unknown rule string? I don't think we need the number of rule IDs for any calculation?

Reply:
That's a good point, I misread the use of threshold, it does seem like it should be calculated per-input.

Reply:
looks right to me

Reply:
We can still skip the call to `collect` though.